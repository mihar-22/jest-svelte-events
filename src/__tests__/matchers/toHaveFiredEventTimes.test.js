import '../../'

import { render, fireEvent } from '@testing-library/svelte'

import Comp from '../fixtures/Comp.svelte'

describe('.toHaveFiredEventTimes', () => {
  test('passes when event has fired correct number of times', async () => {
    const { component, getByText } = render(Comp)
    listen(component, 'e1')
    const button = getByText('e1')
    await fireEvent.click(button)
    expect(component).toHaveFiredEventTimes('e1', 1)
  })

  test('fails when event has not fired correct number of times', async () => {
    const { component, getByText } = render(Comp)
    listen(component, 'e1')
    const button = getByText('e1')
    await fireEvent.click(button)
    expect(() => {
      expect(component).toHaveFiredEventTimes('e1', 2)
    }).toThrowErrorMatchingSnapshot()
  })
})

describe('.not.toHaveFiredEventTimes', () => {
  test('passes when event has not fired number of times', () => {
    const { component } = render(Comp)
    listen(component, 'e1')
    expect(component).not.toHaveFiredEventTimes('e1', 1)
  })

  test('fails when event has fired correct number of times', async () => {
    const { component, getByText } = render(Comp)
    listen(component, 'e1')
    const button = getByText('e1')
    await fireEvent.click(button)
    expect(() => {
      expect(component).not.toHaveFiredEventTimes('e1', 1)
    }).toThrowErrorMatchingSnapshot()
  })
})
