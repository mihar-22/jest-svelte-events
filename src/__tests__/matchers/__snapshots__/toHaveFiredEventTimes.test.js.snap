// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toHaveFiredEventTimes fails when event has fired correct number of times 1`] = `
"[2mexpect([22m[31mcomponent[39m[2m).[22mnot[2m.[22mtoHaveFiredEventTimes[2m([22m[32mexpected[39m[2m)[22m

Expected not to have fired e1: [32m1[39m time
Recevied: [31m1[39m time

Events stack: [31m[\\"e1\\"][39m"
`;

exports[`.toHaveFiredEventTimes fails when event has not fired correct number of times 1`] = `
"[2mexpect([22m[31mcomponent[39m[2m).[22mtoHaveFiredEventTimes[2m([22m[32mexpected[39m[2m)[22m

Expected to have fired e1: [32m2[39m times
Recevied: [31m1[39m time

Events stack: [31m[\\"e1\\"][39m"
`;
